src/HW_models/NRF_RTC.c

```diff
@@ -180,29 +180,53 @@ static void update_all_cc_timers(int rtc) {
   }
 }
 
-void nrf_rtc_timer_triggered() {
-  for ( int rtc = 0; rtc < N_RTC-1/*the 3rd rtc does not have an int*/ ; rtc++ ){
-    if ( RTC_Running[rtc] == false ) {
-      continue;
+static unsigned int get_irq_t(int rtc)
+{
+    unsigned int irq_t = NRF5_IRQ_RTC0_IRQn;
+
+    switch (rtc){
+    case 0:
+      irq_t = NRF5_IRQ_RTC0_IRQn;
+      break;
+    case 1:
+      irq_t = NRF5_IRQ_RTC1_IRQn;
+      break;
+    case 2:
+      irq_t = NRF5_IRQ_RTC1_IRQn;
+      bs_trace_error_line_time("There is no IRQ mapped for RTC2\n");
+      break;
     }
+
+    return irq_t;
+}
+
+static ppi_event_types_t get_event(int rtc)
+{
     ppi_event_types_t event = RTC0_EVENTS_COMPARE_0;
-    unsigned int irq_t         = NRF5_IRQ_RTC0_IRQn;
     switch (rtc){
     case 0:
       event = RTC0_EVENTS_COMPARE_0;
-      irq_t = NRF5_IRQ_RTC0_IRQn;
       break;
     case 1:
       event = RTC1_EVENTS_COMPARE_0;
-      irq_t = NRF5_IRQ_RTC1_IRQn;
       break;
     case 2:
       event = RTC2_EVENTS_COMPARE_0;
-      irq_t = NRF5_IRQ_RTC1_IRQn;
       bs_trace_error_line_time("There is no IRQ mapped for RTC2\n");
       break;
     }
 
+    return event;
+}
+
+void nrf_rtc_timer_triggered() {
+  for ( int rtc = 0; rtc < N_RTC-1/*the 3rd rtc does not have an int*/ ; rtc++ ){
+    if ( RTC_Running[rtc] == false ) {
+      continue;
+    }
+    ppi_event_types_t event = get_event(rtc);
+    unsigned int irq_t      = get_irq_t(rtc);
```

Comment:
I guess we can move this to the place where we use it, and in that way enable using the last RTC to generate PPI events